src/Propel/Runtime/ActiveQuery/InstancePoolTrait.php

```diff
@@ -30,8 +30,9 @@ public static function addInstanceToPool($object, $key = null)
             if ($key === null) {
                 $key = static::getInstanceKey($object);
             }
-
-            self::$instances[$key] = $object;
+            if (!empty($key)) {
```

Comment:
```php
            if (!Propel::isInstancePoolingEnabled()) {
                return;            
            }

            if (!$key) {
                return;
            }
            
            ...
```
Can we use bail early principle and noncloaking instead then?

Reply:
Done